javascript-review.js

```diff
@@ -3,87 +3,227 @@
 
 // What two ways can be used to access the properties of objects in JS? Give examples of both (although assume the objects have been defined).
+// 1) you can access a property of an object using the dot notation, such as obj.property
+// 2) you can access a property of an object using the bracket notation with the key, such as obj["key"]
+
 
 // 5.
 
 // What is JSON? Be clear but concise.
+// JSON is a data format. It stands for Javascript Object Notation. It is represented as a string object and
+// can be useful as a standard format for transmitting data across a network in a client server model.
+
 
 // 6.
 
 // What is a callback?
+// a callback is a higher order function which is passed in as a parameter to another function to be 
+// 'called back' at a later time. Since it is not evaluated until later, we pass in only the function 
+// definition as a parameter and drop the ending brackets. This is possible because functions are objects
+// in Javascript. 
+
 
 // 7.
 
 // Define a function with that takes two number arguments and a callback. The function will add the two numbers and pass them to the callback as an argument. Now call this function and in the callback simply console.log out the argument that has been passed through.
 
+// function callB(arg) {
+//     console.log(arg)
+// }
+
+// function callBackFunc(num1, num22, callB) {
+//     let result = num1 + num22;
+//     callB(result);
+// }
+
+// callBackFunc(5, 8, callB);
+
+
 // 8.
 
 // Why do we use promises?
+// Sometimes we need to get some data from a source, for example a server on a network. This process is not
+// instant and it can take some time to fetch the data. Promises are a way to handle this problem and allow
+// us to have an object which represents this requested data and continue to do some processing in our 
+// program while we wait. Javascript is a single threaded environment but we can process things 
+// asyncronously which can allow us to do things like load one part of a web page before another while we
+// are waiting for some data to come in. 
+// A promise is an object that may produce a single value some time in the future: either a resolved value, or a reason that it’s not resolved.
 
 // 9.
 
 // What are the two methods at our disposal if we have hold of a promise? What are the names of the functions that are related to these functions (that are called when the promise is fulfilled or fails).
+// promise objects have two methods associated with them: 1) resolve(), 2) reject()
+// once we have a promise we can call the .then() method on it to get at what is inside the promise object.
+// if the promise is rejected then we can use the .catch() method to catch the error thrown by the reject() function
 
 // 10.
 
 // What is the DOM?
+// DOM stands for Document Object Model. It refers to the tree like structure of the document object in 
+// in web pages. Each part of the document such as the body or header is represented as a node on the tree
```

Comment:
this is correct but it's not just the document, it includes literally the whole `window` object in the browser